c_src/TLH_absTab.c

```diff
@@ -0,0 +1,225 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <glib.h>
+#include "absTab.h"
+#include "message.h"
+
+struct AbsTab {
+    /* a two-level hash table where the value of each 
+     * key points to another (level2) hash table 
+     */
+    GHashTable *lev1; /* directory watch descriptors */
+    GHashTableIter lev1_rmItr, lev2_rmItr; /* used for remove_next_absTab() */
```

Comment:
I'll defer formatting and styling decision to richard but it does feel weird to see a mix of snake-case and camel-case. I'd suggest picking one and sticking to it.